readthedocs/rtd_tests/tests/test_schema.py

```diff
@@ -0,0 +1,337 @@
+
+    def create_yaml(self, content):
+        file = {
+            'rtd.yml': content,
+        }
+        apply_fs(self.tmpdir, file)
+        return path.join(self.tmpdir.strpath, 'rtd.yml')
+
+    def assertValidConfig(self, content):
+        file = self.create_yaml(content)
+        data = yamale.make_data(file)
+        yamale.validate(self.schema, data)
+
+    def assertInvalidConfig(self, content, msgs=()):
+        file = self.create_yaml(content)
+        data = yamale.make_data(file)
+        with pytest.raises(ValueError) as excinfo:
+            yamale.validate(self.schema, data)
+        for msg in msgs:
+            self.assertIn(msg, str(excinfo.value))
+
+    def test_minimal_config(self):
+        self.assertValidConfig('version: "2"')
+
+    def test_invalid_version(self):
+        self.assertInvalidConfig(
+            'version: "latest"',
+            ['version: \'latest\' not in']
+        )
+
+    def test_invalid_version_1(self):
+        self.assertInvalidConfig(
+            'version: "1"',
+            ['version: \'1\' not in']
+        )
+
+    def test_valid_formats(self):
+        content = '''
+version: "2"
+formats:
+  - pdf
+  - singlehtmllocalmedia
+        '''
+        self.assertValidConfig(content)
+
+    def test_all_valid_formats(self):
+        content = '''
+version: "2"
+formats:
+  - htmlzip
+  - pdf
+  - epub
+  - singlehtmllocalmedia
+        '''
+        self.assertValidConfig(content)
+
+    def test_valid_formats_all(self):
+        content = '''
+version: "2"
+formats: all
+        '''
+        self.assertValidConfig(content)
+
+    def test_invalid_formats(self):
+        content = '''
+version: "2"
+formats:
+  - invalidformat
+  - singlehtmllocalmedia
+        '''
+        self.assertInvalidConfig(
+            content,
+            ['formats', '\'invalidformat\' not in']
+        )
+
+    def tets_empty_formats(self):
+        content = '''
+version: "2"
+formats: []
+        '''
+        self.assertValidConfig(content)
+
+    def test_valid_requirements_file(self):
+        content = '''
+version: "2"
+requirements_file: docs/requirements.txt
+        '''
+        self.assertValidConfig(content)
+
+    def test_invalid_requirements_file(self):
+        content = '''
+version: "2"
+requirements_file: 23
+        '''
+        self.assertInvalidConfig(
+            content,
+            ['requirements_file: \'23\' is not a str']
+        )
+
+    def test_valid_conda(self):
+        content = '''
+version: "2"
+conda:
+  file: environment.yml
+        '''
+        self.assertValidConfig(content)
+
+    def test_invalid_conda(self):
+        content = '''
+version: "2"
+conda:
+   files: environment.yml
+        '''
+        self.assertInvalidConfig(
+            content,
+            ['conda.file: Required']
+        )
+
+    def test_valid_build(self):
+        content = '''
+version: "2"
+build:
+  image: "{image}"
+        '''
+        for image in ['1.0', '2.0', 'latest']:
```

Comment:
You may want to import the valid options from the `-build` package, so we don't need to keep this up to date every time we release a new version.

Reply:
Or maybe we can move the v1 to a similar spec. Also, if we release a new version, isn't better to just support it on the latest spec? I mean, so we can force to the users to update p: